plugins/auto-sizes/tests/test-improve-sizes.php

```diff
@@ -409,4 +409,50 @@ public function test_no_image(): void {
 
 		$this->assertStringContainsString( '<p>No image here</p>', $result );
 	}
+
+	/**
+	 * Test wp_calculate_image_sizes() without improve sizes filter.
+	 */
+	public function test_wp_calculate_image_sizes_without_improve_sizes_filter(): void {
+		// Remove the improve sizes filter.
+		remove_filter( 'wp_content_img_tag', 'auto_sizes_improve_image_sizes_attributes', 9 );
+
+		$image_size = 'large';
+		$image_url  = wp_get_attachment_image_url( self::$image_id, $image_size );
+		$sizes      = wp_calculate_image_sizes( $image_size, $image_url, '', self::$image_id );
+
+		$block_content = '<!-- wp:image {"id":' . self::$image_id . ',"sizeSlug":"' . $image_size . '","linkDestination":"none"} --><figure class="wp-block-image size-' . $image_size . '"><img src="' . wp_get_attachment_image_url( self::$image_id, $image_size ) . '" alt="" class="wp-image-' . self::$image_id . '"/></figure><!-- /wp:image -->';
+
+		$result = apply_filters( 'the_content', $block_content );
+
+		$new_sizes = wp_calculate_image_sizes( $image_size, $image_url, '', self::$image_id );
+
+		$this->assertStringContainsString( $sizes, $result );
+		$this->assertStringContainsString( $new_sizes, $result );
+		$this->assertSame( $new_sizes, $sizes );
+	}
+
+	/**
+	 * Test wp_calculate_image_sizes() with improve sizes filter.
+	 */
+	public function test_wp_calculate_image_sizes_with_improve_sizes_filter(): void {
+		$image_size = 'large';
+		$image_url  = wp_get_attachment_image_url( self::$image_id, $image_size );
+		$sizes      = wp_calculate_image_sizes( $image_size, $image_url, '', self::$image_id );
+
+		$block_content = '<!-- wp:image {"id":' . self::$image_id . ',"sizeSlug":"' . $image_size . '","linkDestination":"none"} --><figure class="wp-block-image size-' . $image_size . '"><img src="' . wp_get_attachment_image_url( self::$image_id, $image_size ) . '" alt="" class="wp-image-' . self::$image_id . '"/></figure><!-- /wp:image -->';
+
+		$result = apply_filters( 'the_content', $block_content );
+
+		$improve_sizes = wp_calculate_image_sizes( $image_size, $image_url, '', self::$image_id );
+
+		$this->assertStringContainsString( $improve_sizes, $result );
+		$this->assertStringNotContainsString( $sizes, $result );
+		$this->assertNotSame( $improve_sizes, $sizes );
+
+		remove_filter( 'wp_content_img_tag', 'auto_sizes_improve_image_sizes_attributes', 9 );
+
+		$new_sizes = wp_calculate_image_sizes( $image_size, $image_url, '', self::$image_id );
+		$this->assertSame( $improve_sizes, $new_sizes );
```

Comment:
why would these match after removing the filter?

some inline docs to explain the tests would be helpful.

Reply:
Thanks. Inline comment added in https://github.com/WordPress/performance/pull/1382/commits/7b2dcf9d96c3fd10c3e6b72b3a59202a01419f2b